### DISCUSSION TITLE
Can I specify which tests should run on remote host versus ones that should run on the local host?

### DISCUSSION THREAD

**Discussion by ifeify** [pytest-dev/pytest-xdist#804] [General]
For example, if I have the following repo structure:
```
src
  |-pkg1
    |- src/
    |- tests/
  |-pkg2
    |- src/
    |- tests/
```

Is there a way to tell `xdist` to run tests from `pkg1` on the local host and tests from `pkg2` on a remote host? The problem I'm trying to solve is `pkg2` is a `pyspark` application and the tests need to run in a glue docker container. `pkg1`, on the other hand, has no special requirements and can run locally. I want to use `pytest-cov` along with `pytest-xdist` so I can get the coverage report for both packages at once. The idea is the glue container will act as the remote host and be ssh-able for `xdist`.

For additional context, for `pkg2`, I'm currently using `coverage` tool but I mount the source repo as a volume in the glue docker container. This way, I can grab the `.coverage` file and combine it with the coverage file from `pkg1` to get the full report after `pytest` runs. The problem here is `coverage run pytest` stops after the first unit test failure and I don't get a full report. `pytest-cov` plugin solves this issue for me and I get a full coverage report regardless of a failing unit test. Just need to make it work with `xdist` so I can test the `pyspark` application also.

Thanks. Looking forward to hearing back.

**Comment by nicoddemus** [comment]
Hi @ifeify,

>  The problem here is coverage run pytest stops after the first unit test failure and I don't get a full report

That's strange, it should just run all tests and report failures. Are you passing `-x` somehow? You should figure out why this is not working.

> Is there a way to tell xdist to run tests from pkg1 on the local host and tests from pkg2 on a remote host?

I don't think so, I would run pytest twice, once for `pkg1` and once for `pkg2`, and then combine the results.
